lib/smart_app_launch/token_exchange_stu2_2_test.rb

```diff
@@ -0,0 +1,10 @@
+module SMARTAppLaunch
+  class TokenExchangeSTU22Test < TokenExchangeTest
+    id :smart_token_exchange_stu2_2
+
+    def make_auth_token_request(smart_token_url, oauth2_params, oauth2_headers)
+      oauth2_headers['Origin'] = Inferno::Application['inferno_host']
```

Comment:
What do you mean when you say public clients? Clients that have no authorization credentials added?

Reply:
http://hl7.org/fhir/smart-app-launch/app-launch.html#support-for-public-and-confidential-apps

Reply:
Okay so I had split out all the cors tests into their own tests, including this one. Which option makes the most sense?

1. Have the CORs token exchange test only run if the client type is public, and omit if it is not public
2. Have all the CORs tests only run if the client type is public, and omit if it is not public
3. Do not have the CORs token exchange test split into it's own test, and instead update the token exchange stu2_2 test shown here in this comment so that it only adds the header if the client type is public, and then a separate CORs test will use that request and check if the correct CORs headers were returned (or omit if the client is not public)

Reply:
I was thinking 3.